_source/_posts/2020-12-02-comparing-authentication-in-laravel-6-7-8.md

```diff
@@ -0,0 +1,202 @@
+
+{%raw%}
+```php
+<a href="{{ url($linkPath) }}">{{ $linkText }}</a>
+```
+{%endraw%}
+
+To display this new component, add `<x-login-link />` in any of your existing Blade templates or layouts.
+
+You can also create components without a template file by rendering them inline. To demonstrate this, replace the `render()` method above with the following:
+
+{%raw%}
+```php
+public function render() {
+    return <<<'blade'
+        <a href="{{ url($linkPath) }}">{{ $linkText }}</a>
+    blade;
+}
+```
+{%endraw%}
+
+While rendering your component inline might be convenient for small components like this, it's better in most cases to keep the Blade template. Separation of concerns between logic and display is one of the primary benefits of using an MVC framework like Laravel.
+
+## Authentication in Laravel 8
+
+I covered all the major updates to Laravel 8 in [this detailed post](/blog/2020/10/15/whats-new-in-laravel-8), but I'll mention two updates here that specifically affect authentication. The first is [Jetstream](https://jetstream.laravel.com/1.x/introduction.html).
+
+Back in 2016, Laravel's creator released [Laravel Spark](https://spark.laravel.com/), which gives you advanced authentication, billing, team management, and security features for a small one-time fee. Jetstream offers many of the same features, but it's now a free part of the Laravel framework ecosystem.
+
+Jetstream is an alternative to the Laravel UI package released in version 6. It adds advanced authentication features like email verification, two-factor authentication, profile photos, account deletion, and integration with Laravel Sanctum, in addition to login and registration. This allows you to start new Laravel applications with less boilerplate code than ever before, and it ensures that these features will be supported by other packages in the Laravel ecosystem.
+
+While Jetstream only supports the [Livewire](https://jetstream.laravel.com/1.x/stacks/livewire.html) or [Intertia.js](https://jetstream.laravel.com/1.x/stacks/inertia.html) frontend frameworks by default, you can publish the scaffolding files and customize them to meet your frontend framework's needs. This will probably be too much work for an existing application, but it might save you hours on setting up your next new Laravel app.
+
+The other change in Laravel 8 that affects authentication is [the addition of an `app/Models/` directory](https://laravel.com/docs/8.x/releases#models-directory). In previous versions of Laravel, all your application's logic lived in subdirectories within the `app/` directory (eg: `app/Console/`, `app/Exceptions`, etc.), but models were tossed right into the `app/` folder directly.
+
+Laravel 8 made this more consistent by moving the default `User.php` file into a new subdirectory. Now a fresh Laravel 8 installation's `app/` directory looks like this:
+
+[Laravel 8 app directory structure](https://i.imgur.com/2sDKv3I.png)
```

Comment:
I would download this image and add it to the post so people don't have to click on it to view it.

Reply:
good catch, not sure how i missed this!